angularSTEP/src/app/discover-display/discover-display.component.ts

```diff
@@ -64,67 +65,77 @@ export class DiscoverDisplayComponent implements OnInit {
+              .get()
+              .then(recipes => {
+                this.recipes = recipes.docs;
+              });
+    } else {
+      switch (this.sortType) {
+        case 'Time Created': {
+          if (this.direction) {
+            this.recipeCollection.ref
+              .orderBy('timestamp', 'asc')
+              .withConverter(new RecipeConverter().recipeConverter)
+              .get()
+              .then(recipes => {
+                this.recipes = recipes.docs;
+              });
+          } else {
+            this.recipeCollection.ref
+              .orderBy('timestamp', 'desc')
+              .withConverter(new RecipeConverter().recipeConverter)
+              .get()
+              .then(recipes => {
+                this.recipes = recipes.docs;
+              });
+          }
+          break;
         }
-        break;
-      }
-      case 'Number of Ingredients': {
-        //temporary until num of recipes are actually added
-        if (this.direction) {
-          this.recipeCollection.ref
-            .orderBy('timestamp', 'asc')
-            .withConverter(new RecipeConverter().recipeConverter)
-            .get()
-            .then(recipes => {
-              this.recipes = recipes.docs;
-            });
-        } else {
-          this.recipeCollection.ref
-            .orderBy('timestamp', 'desc')
-            .withConverter(new RecipeConverter().recipeConverter)
-            .get()
-            .then(recipes => {
-              this.recipes = recipes.docs;
-            });
+        case 'Rating': {
+          //temporary until num of recipes are actually added
+          if (this.direction) {
+            this.recipeCollection.ref
+              .orderBy('rating', 'desc')
+              .withConverter(new RecipeConverter().recipeConverter)
+              .get()
+              .then(recipes => {
+                this.recipes = recipes.docs;
+              });
+          } else {
+            this.recipeCollection.ref
+              .orderBy('rating', 'asc')
+              .withConverter(new RecipeConverter().recipeConverter)
+              .get()
+              .then(recipes => {
+                this.recipes = recipes.docs;
+              });
+          }
+          break;
         }
-        break;
-      }
-      case 'Name': {
-        if (this.direction) {
-          this.recipeCollection.ref
-            .orderBy('recipeName', 'desc')
-            .withConverter(new RecipeConverter().recipeConverter)
-            .get()
-            .then(recipes => {
-              this.recipes = recipes.docs;
-            });
-        } else {
-          this.recipeCollection.ref
-            .orderBy('recipeName', 'asc')
-            .withConverter(new RecipeConverter().recipeConverter)
-            .get()
-            .then(recipes => {
-              this.recipes = recipes.docs;
-            });
+        case 'Name': {
+          if (this.direction) {
```

Comment:
Overall seems like there's a lot of copied and pasted code here with just a few spots tweaked for each sort option. Could the switch/case work out which parts need to be tweaked and then all this long code only needs to be written+maintained once?

Reply:
Wow yes, I could have DEFINITELY made that easier! Thanks, so much cleaner and simpler now.